### ISSUE TITLE
[ONNX] Exporting with `dynamo=True` and `Dim.DYNAMIC` in `dynamic_shapes` passes for `scaled_dot_product_attention`, but doesn't do anything

### ISSUE THREAD

**Issue by cyanic-selkie** [pytorch/pytorch#152018]
### 🐛 Describe the bug

Basically, I'm trying to export a custom transformer model that uses `scaled_dot_product_attention` op to implement the attention layers, but I can't properly export it. Like the title says, setting `dynamic_shapes` appears to work (i.e., nothing crashes), but the exported model's input's/output's dimensions are specialized.

In the example below, simply returning the embeddings instead of the SDPA output works fine. It also works fine if I'm not using dynamo.

```python
class Model(nn.Module):
    def __init__(self):
        super().__init__()

        self.embeddings = nn.Embedding(1000, 512)

    def forward(
        self,
        input_ids: torch.Tensor,
    ) -> torch.Tensor:
        embeddings = self.embeddings(input_ids)

        return torch.nn.functional.scaled_dot_product_attention(
            embeddings,
            embeddings,
            embeddings,
        )


model = Model()

torch.onnx.export(
    model,
    (torch.randint(0, 1000, (16, 128), dtype=torch.long),),
    "decoder.onnx",
    dynamo=True,
    dynamic_shapes=((Dim.DYNAMIC, Dim.DYNAMIC),),
)
```

### Versions

PyTorch version: 2.6.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 15.3.2 (arm64)
GCC version: Could not collect
Clang version: 17.0.0 (clang-1700.0.13.3)
CMake version: version 4.0.0
Libc version: N/A

Python version: 3.12.0 (main, Oct  2 2023, 20:56:14) [Clang 16.0.3 ] (64-bit runtime)
Python platform: macOS-15.3.2-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Pro

Versions of relevant libraries:
[pip3] Could not collect
[conda] Could not collect

(I am using `uv`, the `torch` version is 2.6.0, `onnx` is 1.17.0 and `onnxruntime` is 1.20.1)

**Comment by justinchuby** [comment]
Please test with `torch.onnx.export(..., dynamo=True, report=True)` using the latest torch-nightly. Attach the generated report if there is an error. Thanks!

**Comment by cyanic-selkie** [comment]
@justinchuby Since 2.17 was just released, I tried it and it worked, thanks!
